### PR TITLE
docs: Update post-release checklist

### PR DESCRIPTION
**Pull request by webvictim**
Updates the post-release checklist to remove unnecessary items and reorganise.

### CODE REVIEW

File: docs/postrelease.md [hunk 1]
```diff
@@ -5,17 +5,14 @@ This checklist is to be run after cutting a release.
...
-  - Example: https://github.com/gravitational/teleport/pull/4608
-- [ ] Create PR to update app version and Helm chart version (https://github.com/gravitational/teleport/blob/master/examples/chart)
+  - Example command: `TELEPORT_VERSION=6.2.0 make -C assets/aws create-update-pr`
 
-### Major/minor releases only
+### Major releases only
 
 - [ ] Update support matrix in docs FAQ page
-- [ ] Create PR to update default Docker image version in Teleport docs
-  - Example: https://github.com/gravitational/teleport/pull/4615
```
Comment (wadells) [inline docs/postrelease.md:17]: Is this no longer needed, or handled by a sibling step?

**Review by wadells — APPROVED** [review]

File: docs/postrelease.md (hunk 1 above)
Comment (webvictim) [reply docs/postrelease.md:17]: It's now handled with a [variable include in the docs](https://github.com/gravitational/teleport/blame/branch/v6.2/docs/pages/quickstart-docker.mdx#L31-L34) from the [`docs/config.json` file](https://github.com/gravitational/teleport/blob/branch/v6.2/docs/config.json#L231-L240).

**Review by quinqu — APPROVED** [review]

**Review by inertial-frame — APPROVED** [review]
QQ - Might be a silly question: whose responsibility is it to update the versioning for Docs? 

W.R.T. - https://github.com/gravitational/teleport/pull/7033 and https://github.com/gravitational/next/pull/269

**Comment by webvictim** [comment]
@inertial-frame I'm actually not sure. Traditionally this was handled by the product manager but I think it makes sense to have it done by the release engineer (and should be added to the list of things that need automating)

**Pull request merged**